deploy/kubernetes/driver/kubernetes/README.md

```diff
@@ -0,0 +1,33 @@
+# CSI Driver deployment developer notes for IBM Cloud Kubernetes service (IKS)
+
+## Namespaces
+
+In general, the following namespaces are used:
+
+| Object Kind | Namespace |
+| --- | --- |
+| Kustomization | kube-system |
+| ConfigMap | kube-system |
+| ServiceAccount | default |
```

Comment:
why service account need to be on default ns? 

can we have namespace as configurable for all resources, this way we need not to maintain two set of deployment or so.

Reply:
Original codeline for IKS used `kube-system` and `default`, this just documents the configuration for future developers to reference; and in the code, I have appended specific declaration for `default` instead of omitted namespace key in the YAML objects.

It is possible to have namespace in Kustomize overlay files that are relatively static, or instructing the end-user to potentially run `kustomize edit set namespace NAME_HERE`. However for IKS it will always be static, for OCP Self-Managed it will also be static.

There is enough divergence between IKS and OCP, that having separate subdirectories will (at least in the short-term) aid usability and reduce mistakes.